### DISCUSSION TITLE
Type-trait to get scalar type of a nested array type.

### DISCUSSION THREAD

**Discussion by frettini** [mitsuba-renderer/mitsuba3#1262] [Q&A] [answered]
Hello everyone, 
Does Mitsuba offer a type-trait (or another method) to get the scalar type of a nested array type? For example, I would like to retrieve `ScalarVector3f` for a given `Vector3f`. I found drjit's `scalar_t` type-trait, but this will return the scalar type of the nested array.
If it doesn't exist, is it in practice possible to implement such a trait? 
Many thanks.

**Comment by njroussel** [comment, marked as answer]
Hi @frettini 

No, we don't have such a trait currently. This will be improved in an upcoming version though: [we'll have a `dr.replace_type_t`](https://drjit.readthedocs.io/en/latest/reference.html#drjit.replace_type_t).

For now you'd be stuck with either some hacky python thing like `mi.getattr("Scalar" + original_type.__name__)` or some extra C++ bindings. Depending on what you need, you might as well just make a big switch statement - there aren't that many possibilites.

**Comment by frettini** [comment]
Hi @njroussel , 
Thank you for your reply and apologies for the delay. In the end, I went for another approach that didn't necessitate this. Good to know about the changes in the upcoming version though! Thanks
